tests/test_api.py

```diff
@@ -1055,3 +1056,38 @@ def test_open_array_with_mode_r_plus(store: Store) -> None:
     assert isinstance(z2, Array)
     assert (z2[:] == 1).all()
     z2[:] = 3
+
+
+@pytest.mark.parametrize("func_name", zarr.api.synchronous.__all__)
+def test_derived_docstrings(func_name: str) -> None:
+    """
+    Test that functions in the synchronous API module have
+    docstrings that are derived from those in the asynchronous API module
+    """
+    if func_name in ("open_array",):
+        msg = (
+            '"open_array" gets xfailed because the async version and the sync '
+            "version have meaningful differences in their docstrings."
+        )
+        pytest.xfail(reason=msg)
+    assert (
+        getattr(zarr.api.synchronous, func_name).__doc__
+        == getattr(zarr.api.asynchronous, func_name).__doc__
+    )
+
+
+@pytest.mark.parametrize("func_name", zarr.api.synchronous.__all__)
+def test_derived_signatures(func_name: str) -> None:
+    """
+    Test that functions in the API module have signatures that are derived from those in the asynchronous API module
+    """
+    if func_name in ("tree", "load", "open", "open_array", "group"):
+        msg = (
+            f"{func_name} gets xfailed because the async version and the sync "
+            "version have meaningful differences in their signatures."
+        )
+        pytest.xfail(reason=msg)
+
+    sync_sig_params = inspect.signature(getattr(zarr.api.synchronous, func_name)).parameters
```

Comment:
Is it worth checking [`Signature.return_annotation`](https://docs.python.org/3/library/inspect.html#inspect.Signature.return_annotation) here too?